source/Ship.h

```diff
@@ -358,13 +362,17 @@ class Ship : public Body, public std::enable_shared_from_this<Ship> {
 	// Check if this is a ship of a type that can be carried.
 	bool CanBeCarried() const;
 	// Move the given ship into one of the bays, if possible.
-	bool Carry(const std::shared_ptr<Ship> &ship);
+	bool Carry(const std::shared_ptr<Ship> &ship, std::vector<Visual> *visuals = nullptr);
```

Comment:
Let's drop the default parameter

```suggestion
	bool Carry(const std::shared_ptr<Ship> &ship, std::vector<Visual> *visuals);
```

I think ideally we would have two methods, one called from external contexts with visuals available and the other called from an internal context where visuals do not make sense to display (such as the preflight loading of carried ships post-recharge), but there is some additional refactoring needed to reach that state.